velox/dwio/parquet/reader/ParquetData.h

```diff
@@ -35,14 +35,18 @@ class ParquetParams : public dwio::common::FormatParams {
   ParquetParams(
       memory::MemoryPool& pool,
       dwio::common::ColumnReaderStatistics& stats,
-      const FileMetaDataPtr metaData)
-      : FormatParams(pool, stats), metaData_(metaData) {}
+      const FileMetaDataPtr metaData,
+      const std::string sessionTimezone)
+      : FormatParams(pool, stats),
+        metaData_(metaData),
+        sessionTimezone_(std::move(sessionTimezone)) {}
   std::unique_ptr<dwio::common::FormatData> toFormatData(
       const std::shared_ptr<const dwio::common::TypeWithId>& type,
       const common::ScanSpec& scanSpec) override;
 
  private:
   const FileMetaDataPtr metaData_;
+  const std::string sessionTimezone_;
```

Comment:
At this level, you should have some object more efficient to apply tz conversion instead of a string, which would require some potential global lock when acquiring such an object.  Do you have a follow up PR for how the tz conversion will be done?  We probably should put them in the same PR.

Reply:
@Yuhta The purpose of adding timezone information in the Parquet `PageReader` is to support int64 timestamps with `isAdjustedToUTC=0`. When processing it, the timestamp needs to be converted to UTC time based on the local timezone.

The follow up PR depends on the merging of the Parquet timestamp reader PR https://github.com/facebookincubator/velox/pull/8325. In #8325, it only supports `isAdjustedToUTC=1`.

In our internal repository, I use the timezone string to convert it to a timezone id, and then call `Timestamp::toGMT(timezone id)` to convert the timestamp to UTC.
```cpp
sessionTzID = util::getTimeZoneID(sessionTzName);
timestamp.toGMT(sessionTzID);
```

Reply:
I think the best way is to do the conversion from string to `const date::time_zone*` in data source, and then we do not need to do the conversion for each split.  In split reader and file option we only need keep the `const date::time_zone*` and pass it to `Timestamp::toGMT` for each row.